components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/StandaloneJobOrchestrator.cs

```diff
@@ -170,6 +170,24 @@ IJsonSerializer jsonSerializer
 
                         if (diagInfo.Id == assignedJob.Value.Job.Id) {
                             _publisherDiagnosticInfo.AddOrUpdate(assignedJob.Value.Job.Id, diagInfo);
+
+                            foreach (var entry in _publishedNodesEntries) {
+                                var id = _publishedNodesJobConverter.
+                                    ToConnectionModel(entry, _standaloneCliModel).CreateConnectionId();
+
+                                if (_publisherDiagnosticInfo[diagInfo.Id].Id == id) {
+                                    _publisherDiagnosticInfo[id].EndpointInfo = new PublishedNodesEntryModel {
```

Comment:
The main diagnostic reporting logic is not using `EndpointInfo` at all. We only need it here where we try to find corresponding `PublishedNodesEntryModel` object for each `JobDiagnosticInfoModel` object. So I would suggest to remove `EndpointInfo` field from `JobDiagnosticInfoModel` and instead change `_publisherDiagnosticInfo` to be a dictionary from string `jobId` to Tuple<JobDiagnosticInfoModel, PublishedNodesEntryModel>.

This way we can keep the information about corresponding objects an not require cross-project dependencies that were introduced in the PR. In this case the return type from `GetDiagnosticInfoAsync()` would change to `List<Tuple<JobDiagnosticInfoModel, PublishedNodesEntryModel>>` and then the `ToApiModel()` transformation method can take care of changing those into `DiagnosticInfoApiModel`.

Reply:
removed `EndpointInfo` from `JobDiagnosticInfoModel`.
I also remove the  cross-project dependencies. 
`_publisherDiagnosticInfo` is now  `Dictionary<string, Tuple<JobDiagnosticInfoModel, PublishedNodesEntryModel>>`.
Using a Dictionary instead of List I was able to keep all the dictionary properties used to manage the JobDiagnosticInfoModel